src/PgNamed.hs

```diff
@@ -262,8 +267,11 @@ executeNamed
     -> [NamedParam]   -- ^ The list of named parameters to be used in the query
     -> m Int64        -- ^ Number of the rows affected by the given query
 executeNamed conn qNamed params =
-    withNamedArgs qNamed params >>= \(q, actions) ->
-        liftIO $ PG.execute conn q (toList actions)
+    withNamedArgs qNamed params >>= \(q, actions) -> do
+        res <- liftIO $ try $ PG.execute conn q (toList actions)
```

Comment:
There are other places can potentially throw SQLError:
E.g. search for `PG.queryWith`, `PG.query`

Reply:
This can probably become helper function that replaces all usages of liftIO within the module.